src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/PiglinListener.java

```diff
@@ -38,6 +48,30 @@ public void onEntityPickup(EntityPickupItemEvent e) {
         }
     }
 
+    @EventHandler
+    public void onPiglinDropItem(EntityDropItemEvent e) {
+        if (e.getEntity() instanceof Piglin) {
+            Piglin piglin = (Piglin) e.getEntity();
+            Set<ItemStack> drops = SlimefunPlugin.getRegistry().getBarterDrops(); 
+            
+            /*
+             * NOTE:  Getting a new random number each iteration because multiple items could have the same
+             * % chance to drop, and if one fails all items with that number will fail.  Getting a new random number
+             * will allow multiple items with the same % chance to drop.
+             */
+            
+            for (ItemStack is : drops) {
+                SlimefunItem sfi = SlimefunItem.getByItem(is);
+                if (sfi instanceof PiglinBarterDrop && ((PiglinBarterDrop)sfi).getBarteringLootChance() >= ThreadLocalRandom.current().nextInt(1,101)) {
+                    Item drop = piglin.getWorld().dropItemNaturally(piglin.getEyeLocation(), sfi.getItem());
+                    drop.setVelocity(e.getItemDrop().getVelocity());
+                    e.getItemDrop().remove();
```

Comment:
You could also just do `e.getItemDrop().setItemStack(sfi.getItem().clone())` here, this would probably cause less issues with other plugins.